crates/client/eth/src/state_update.rs

```diff
@@ -0,0 +1,212 @@
+        contract DummyContract {
+            event LogStateUpdate(uint256 globalRoot, int256 blockNumber, uint256 blockHash);
+
+            function fireEvent() public {
+                uint256 globalRoot = 2814950447364693428789615812443623689251959344851195711990387747563915674022;
+                int256 blockNumber = 662703;
+                uint256 blockHash = 1119674286844400689540394420005977072742999649767515920196535047615668295813;
+
+                emit LogStateUpdate(globalRoot, blockNumber, blockHash);
+            }
+        }
+    );
+
+    const ETH_URL: &str = "https://eth.merkle.io";
+    const L1_BLOCK_NUMBER: u64 = 20395662;
+    const L2_BLOCK_NUMBER: u64 = 662703;
+    const EVENT_PROCESSING_TIME: u64 = 2; // Time to allow for event processing in seconds
+
+    /// Test the event subscription and state update functionality
+    ///
+    /// This test performs the following steps:
+    /// 1. Sets up a mock Ethereum environment using Anvil
+    /// 2. Initializes necessary services (Database, Metrics)
+    /// 3. Deploys a dummy contract and sets up an Ethereum client
+    /// 4. Starts listening for state updates
+    /// 5. Fires an event from the dummy contract
+    /// 6. Waits for event processing and verifies the block number
+    #[tokio::test]
+    async fn listen_and_update_state_when_event_fired_works() {
+        // Set up chain info
+        let chain_info =
+            ChainInfo { chain_id: Felt::from_dec_str("11153111").unwrap(), chain_name: "testing".to_string() };
+
+        // Set up database paths
+        let temp_dir = TempDir::new().expect("issue while creating temporary directory");
+        let base_path = temp_dir.path().join("data");
+        let backup_dir = Some(temp_dir.path().join("backups"));
+
+        // Initialize database service
+        let db = Arc::new(
+            DatabaseService::new(&base_path, backup_dir, false, &chain_info)
+                .await
+                .expect("Failed to create database service"),
+        );
+
+        // Set up metrics service
+        let prometheus_service = MetricsService::new(true, false, 9615).unwrap();
+        let block_metrics = BlockMetrics::register(&prometheus_service.registry()).unwrap();
+
+        // Set up Anvil for local Ethereum testing
+        let anvil = Anvil::new()
+            .fork(ETH_URL)
+            .fork_block_number(L1_BLOCK_NUMBER)
+            .try_spawn()
+            .expect("Failed to spawn Anvil instance");
+        let rpc_url: Url = anvil.endpoint().parse().expect("Failed to parse RPC URL");
+
+        // Set up Ethereum provider and contracts
+        let provider = ProviderBuilder::new().on_http(rpc_url.clone());
+        let contract = DummyContract::deploy(provider.clone()).await.unwrap();
+        let core_contract = StarknetCoreContract::new(contract.address().clone(), provider.clone());
+
+        let eth_client = EthereumClient { provider: Arc::new(provider), l1_core_contract: core_contract.clone() };
```

Comment:
actually, can we not import the fixture from eth client directly?

Reply:
anvil fixture added